### PR TITLE
[API] Added current fork status getter API

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

The current Klaytn implementation only knows the current fork status through predefined comparison internally. This PR proposes explicitly showing the current fork-update status through an API getter call.

Using this API, users can easily query the fork status with negligible management costs. This function should be updated whenever a future fork is scheduled. 

Sample output:
```
> klay.forkStatus(20)
{
  EthTxType: true,
  Istanbul: true,
  KIP103: false,
  Kore: true,
  London: true,
  Magma: true,
  Mantle: false
}
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kjeom** [comment]
How about just add the hard fork status info. to the `klay_clientVersion` API ?

**Comment by hyunsooda** [comment]
Hi @kjeom, that seems like another appropriate position to place this change. However, the node package does not have an exported interface to access the required values, such as getting the current block number due to the well-defined orgnization. Any other suggestions? If this change is anticipated to be useless, another option is to close it.

**Comment by aidan-kwon** [comment]
@hyunsooda How about returning true/false for each hardforks? All hardforks are not designed to be a squeal update. A chain may skip some hardforks such as KIP103Fork.

**Comment by hyunsooda** [comment]
@aidan-kwon, Updated. Please give further feedback if you have.

**Review by aidan-kwon — DISMISSED** [review]
Thanks! Please update this on the Klaytn Docs API page 
cc @kjeom

**Comment by blukat29** [comment]
@hyunsooda Can we have explicit `false` for inactive forks? I'd like to see the list of forks. e.g.
```
klay.forkStatus(20)
{
  EthTxType: true,
  Istanbul: true,
  Kore: false,
  KIP103: false,
  London: true,
  Magma: true
}
```

**Comment by hyunsooda** [comment]
@blukat29, Sure.

TL;DR
I believe that by leveraging the robust type system of Golang, specifically through the reflect package, it is possible to refactor this convoluted if-else code into a more advantageous solution. This approach eliminates the need to update the function in the future if the struct field members have specific rules, such as requiring the block number to have a suffix of Block. However, upon closer consideration, I realized that implementing such implicit rules might introduce unforeseen side effects, particularly when modifications are made to the ChainConfig structure.

File: api/api_public_blockchain.go [hunk 1]
```diff
@@ -150,6 +150,53 @@ func (s *PublicBlockChainAPI) GetAccount(ctx context.Context, address common.Add
...
+	}
+	blockNumber := block.Number()
+	cfg := s.b.ChainConfig()
+	activated := make(map[string]bool)
+
+	if cfg.IsIstanbulForkEnabled(blockNumber) {
+		activated["Istanbul"] = true
+	} else {
+		activated["Istanbul"] = false
+	}
```
Comment (blukat29) [inline api/api_public_blockchain.go:166]: How about this?

```go
activated := map[string]bool{
  "Istanbul": cfg.IsIstanbulForkEnabled(blockNumber),
  "London": cfg.IsLondonForkEnabled(blockNumber),
}
```

Comment (hyunsooda) [reply api/api_public_blockchain.go]: Ah..?

**Review by aidan-kwon — APPROVED** [review]

**Review by blukat29 — APPROVED** [review]

**Comment by aidan-kwon** [comment]
@kjeom @JayChoi1736 @nohkwak Please, take a look

**Review by JayChoi1736 — APPROVED** [review]

**Comment by aidan-kwon** [comment]
@kjeom Can you review this?

**Comment by kjeom** [comment]
note: we should f/u this API every hard fork.

@hyunsooda I recommend you to add an function that return that information in the `params/config.go` because it is easy to miss when we are on hardfork update

**Comment by aidan-kwon** [comment]
> note: we should f/u this API every hard fork.
> 
> @hyunsooda I recommend you to add an function that return that information in the `params/config.go` because it is easy to miss when we are on hardfork update

@kjeom  It would be better to make a test case checking a remainder after parsing the API

**Comment by hyunsooda** [comment]
@kjeom,

>  I recommend you to add an function that return that information in the params/config.go because it is easy to miss when we are on hardfork update

Do you mean that it's better to move the function in `param/config.go`? If not, could you elaborate more? I couldn't exactly get what your concern is.

@aidan-kwon, 

I think that rewriting this proposed API with the result of `Rules()`. And, the excpetional two cases (`EthTxType`, `KIP103`) can be manually added. Then, I think we don't need to worry about the missing fork update. While I was writting this, you merged this. How do you think?
